# Profitable App Profiles for the App Store and Google Play Markets

The objective of the project is to evaluate the profitable app profiles for both App Store and Google Play markets for development.

Given that the goal is to increase the revenue from in-app ads, the analysis will be focused on apps that are:
- Free
- Interested by a large number of users to download
- Popular in both App Store and Google Play markets

# Explore the data set

Two sets of data available for public will be used in this analysis:
- Google Play: A data set with approximately 10 thousand Android apps (<Link of data set>)
- App Store: A data set with approximately 7 thousand iOS apps (Link of data set)

In [1]:
from csv import reader

#open dataset of Google Play
open_file =  open('googleplaystore.csv')
read_file = reader(open_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

#open dataset of App Store
open_file =  open('AppleStore.csv')
read_file = reader(open_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In order to explore the data sets easier, the following `explore_data()` function is created to slice the data sets into separated rows, with an option to display the number of rows and columns in the data sets. By default, this option is set to not displaying.

In [2]:
#creating explore_data function
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

First, we will take a look at the values in the data sets.

In [40]:
explore_data(android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10840
Number of columns: 13


In [42]:
explore_data(ios, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


Next, we will take a look at the names of the headers in the data sets.

In [4]:
#checking the names of the columns
print(android_header)
print(ios_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The following is the basic information of the data sets:
- Google Play data set: There are 10840 entries and 13 columns
- App Store data set: There are 7197 entries and 16 columns

## Cleaning and organizing the data for analysis

Before the analysis, the data in the data sets are cleaned and organized to make sure that it is ready for analysis purposes. While investigating the data sets, there are some issues discovered including wrong entires and duplicated entries of data.

In the following steps, we are going to clean the data set for use.

### Removing wrong entry in Andriod app data set

From the discussion forum of other data set users, we've discovered that there are wrong entry of data in the data set of Android apps.

In this step, we are going to remove the wrong entries in the data sets.

In [5]:
#find the row of the missing datapoint
for row in android: 
    if len(row) != len(android_header):
        print(row)
        print(android.index(row))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10472


In [6]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [7]:
#delete the row with error
del android[10472]

In [8]:
print(android[10472])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


We've removed the entry with the wrong data in Android app data set.

### Removing duplicated app entries in Android list

We are going to remove the duplicated entries (entries that contain the same apps) to ensure that we capture the relevant characteristics and value of an individual app. 

For example, if an app have multiple entries, we would want to make sure that we capture only the characterisitic that are the most representative for that specific app.

From the investigation of the list, it seems that only Android app data set contains significant duplicates but not the iOS app data set.

In the following step, we are going to remove the duplicates in the Android app data set.

In [9]:
#check for some examples of duplicated app entries
for app in android:
    name = app[0]
    
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


From the example above, we can see that a single app, i.e. Instagram, contains 4 different entries.

In [10]:
duplicate_apps_android = []
unique_apps_android = []

for app in android:
    name = app[0]
    if name in unique_apps_android:
        duplicate_apps_android.append(name)
    else:
        unique_apps_android.append(name)
    
print('Number of duplicate apps:', len(duplicate_apps_android))
print('Examples of duplicate apps:', duplicate_apps_android[:15])

Number of duplicate apps: 1181
Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


There are 1181 duplicates in the Android app list.

Next, we are going to remove these duplicates from the Android list. Since we are going to evaluate the popularity of the apps, we are going to keep the entry with the highest number of reviews. The higher number of reviews can also represent that latest entry in the data set.

In [11]:
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

print(len(reviews_max))

9659


In [12]:
print('Expected length: ', len(android) - 1181)
print('Actual length: ', len(reviews_max))

Expected length:  9659
Actual length:  9659


In [13]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name) #to keep track of apps that we already added        

In [14]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


After removing the duplicates, there are 9659 number of entries in the Android app data set for analysis.

## Filtering out apps that are irrelevant in the analysis

Since the app development will be focused on English apps and free apps, in the following steps, we are going to filter out apps that are irrelevant for this analysis.

### Removing the non-English apps

In this step, we are going to exclude non-English apps in the data sets. The following are some examples of non-English apps in the data sets.

In [15]:
print(ios[813][1])
print(ios[6731][1])
print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


First, in order to check whether the apps contain non-English characters, we are going to build a function `is_english()` to check the encoding number of each character in the app names.

In [16]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True
#return True should not be put in the loop in case there are some english characters in the name of the app
#which may contain both other characters and english characters

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
False
False


As shown above, the function cannot correctly identify certain English app names that contrain special characters, we set a criteria to include these apps under the following condition:

- English apps with up to three emoji or other special characters are labeled as English apps

We are going to modify the function `is_english()` to include the condition mentioned.

In [17]:
def is_english(string):
    count = 0
    
    for character in string:

        if ord(character) > 127:
            count += 1
        
    if count > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
True
False


There are still some non-English apps that pass through the filer, but it will be good enough at this point for our analysis.

Below, we are going to use `is_english()` to create a list that only contain English apps.

In [18]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name) == True:
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name) == True:
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

After filtering out the non-English apps, there are 9614 Android apps and 6183 iOS apps for our analysis.

### Isolating free apps

Next, we are going to exclude paid apps from our data sets to have clean data sets of free Android apps and iOS apps for our analysis.

In [44]:
android_final = []

for app in android_english:
    price = app[7]
    
    if price == '0':
        android_final.append(app)
        
print(len(android_final))

8864


In [45]:
ios_final = []

for app in ios_english:
    price = app[4]
    
    if price == '0.0':
        ios_final.append(app)
        
print(len(ios_final))

3222


After cleaning the data, there are 8864 Android apps and 3222 iOS apps for our analysis.

## Analyzing the data sets

After cleaning and organizing the data, we are going to analyze the apps by genres by evaluating the most common genres and the number of downloads in both Google Play and App Store markets.

We can have a prelimiary understanding of:
- Market competition by looking at the number of apps in the genres in the market
- Users' interest by looking at the number of downloads or reviews of the apps by genres

### Validation strategy for an app idea

To minimize risks and overhead, our validation strategy for an app idea has three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

In this case, we need to find app profiles that are successful in both markets.

### Most common apps by genre by design

In order to ease our analysis, we are going to build two functions as below:
- `freq_table()`: To genereate frequency tables in a dictionary data type
- `display_table()`: To display the frequency table generated with sorted values of percentages

In [21]:
#a function to generate frequency table
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1 #count the number of values
        value = row[index]
        
        if value in table:
            table[value] += 1 #dictionary value
        else:
            table[value] = 1
            
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
    
    return table_percentages

In [22]:
#another function to display the frequency table of the columns
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

First, we are going to investigate the common genres in App Store.

We are going to examine the `prime_genre` column in the iOS data set.

In [23]:
display_table(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


From the preliminary analysis, the most common genre is Games (58%), followed by Entertainment (8%), Photo & Video (5%), Education (4%) and Social Networking (3%).

It seems that most free ios apps are designed for entertainment purpose such as social networking, shopping, sports, music, etc. rather than for practical purposes such as productivity, lifestyle, business, etc. However, the frequency table alone is not sufficient for the recommendation because it purely indicates the number of apps designed, not the number of apps adopted by users, which is the number of users of the apps.

Second, we are going to investigate the common categories and genres in Google Play.

We are going to examine the `Category` column.

In [24]:
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

We are going to examine the `Genres` column.

In [25]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

From prelimiary analysis, it seems that there are two columns indicating the genres of the android apps (`Category` and `Genres`) and their differences are not clear.

1. From category column, we can see that Family is the dominating category (18%) compared to the others, yet the dominating effect is not as obvious as genre Games in App Store.

2. From genres column, we can see a more balanced results between different genres, where the most common ones are Tools (8%), Entertainment (6%), Education (5%) and Business (5%).

Further analysis will have to be conducted to derive a conclusion with higher confidence level.

### Most common apps by number of downloads

From the previous analysis, the number of apps by categories are shown where it is not sufficient to demonstrate the number of apps installed or used by users.

In the following steps, we are going to analayze the number of apps by genres / categories installed by users for both App Store and Google Play.

#### Most common apps installed in App Store

Since there is no information of the number of installations, we are going to look at the number of user ratings to determine the popularity of the apps.

In [26]:
ios_genre = freq_table(ios_final, -5)

for genre in ios_genre:
    total = 0
    len_genre = 0
    
    for app in ios_final:
        genre_app = app[-5]
        
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
            
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


From the result above, genres of Navigation, Reference and Social Networking have remarkably larger number of users compared to the rest of the genres. We might want to take a deeper look into these two genres.

In [27]:
#further investigate what is in the navigation genre
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


We see that there are two apps, i.e., Waze - GPS Navigation, Maps & Real-time Traffic and Google Maps, dominating the number of ratings given.

In [29]:
#further investigate what is in reference genre
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


We see that there is one app - Bible, dominating the number of ratings in the reference genre.

In [30]:
#further investigate what is in social networking genre
for app in ios_final:
    if app[-5] == 'Social Networking':
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

We can see that the dominating social media apps such as Facebook and Pinerest have remarkably larger number of user ratings compared to the result of the apps in the genre.

By comparing the genres of apps designed and the genres of apps downloaded, it seems that there is not a prominent pattern to draw a conclusion of the most popular genre of apps to build in App Store.

By looking at the number of user ratings, we could try to determine the interest of the users in different genres. By looking at the number of genres in App Store, we could try to determine the level of competition in the iOS app market.

In this case, genres with lower competition and higher user interest would be more appropriate to develop, these include:
1. Music 
2. Book
3. Finance
4. Food & Drink
5. Travel
6. Photo & Video
7. Shopping

P.S We also take into account that some genres already have dominating apps where it will take more effort to change behavioral habit of the users.

#### Most common apps installed in android market

We are going to look at the number of installations to determine the popularity of the apps.

In [46]:
android_category = freq_table(android_final, 1)

for category in android_category:
    total = 0
    len_category = 0
    
    for app in android_final:
        category_app = app[1]
        
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_category += 1
            
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

From the analysis above, there are a few genres that have much higher number of installations compared to the rest:
- COMMUNICATION
- ENTERTAINMENT
- GAME
- SOCIAL
- PHOTOGRAPHY
- TRAVEL_AND_LOCAL
- TOOLS
- PRODUCTIVITY
- VIDEO_PLAYERS
- NEWS_AND_MAGAZINES
- SHOPPING

We can take a deeper look into some of them to see if there are dominating apps in the category.

In [33]:
#further investigate what is in communication
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+' or
                                     app[5] == '500,000,000+' or
                                     app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [34]:
#further investigate what is in social
for app in android_final:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+' or
                                     app[5] == '500,000,000+' or
                                     app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


In [35]:
#further investigate what is in entertainment
for app in android_final:
    if app[1] == 'ENTERTAINMENT' and (app[5] == '1,000,000,000+' or
                                     app[5] == '500,000,000+' or
                                     app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Hotstar : 100,000,000+
Talking Angela : 100,000,000+
IMDb Movies & TV : 100,000,000+
Talking Ben the Dog : 100,000,000+
Netflix : 100,000,000+


In [36]:
#further investigate what is in tools
for app in android_final:
    if app[1] == 'TOOLS' and (app[5] == '1,000,000,000+' or
                                     app[5] == '500,000,000+' or
                                     app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google : 1,000,000,000+
Google Translate : 500,000,000+
Calculator : 100,000,000+
Device Help : 100,000,000+
Account Manager : 100,000,000+
SHAREit - Transfer & Share : 500,000,000+
Samsung Calculator : 100,000,000+
Gboard - the Google Keyboard : 500,000,000+
Google Korean Input : 100,000,000+
Share Music & Transfer Files - Xender : 100,000,000+
Tiny Flashlight + LED : 100,000,000+
GO Keyboard - Cute Emojis, Themes and GIFs : 100,000,000+
Speedtest by Ookla : 100,000,000+
CM Locker - Security Lockscreen : 100,000,000+
Applock : 100,000,000+
Clean Master- Space Cleaner & Antivirus : 500,000,000+
Lookout Security & Antivirus : 100,000,000+
Google Now Launcher : 100,000,000+
360 Security - Free Antivirus, Booster, Cleaner : 100,000,000+
Samsung Smart Switch Mobile : 100,000,000+
Avast Mobile Security 2018 - Antivirus & App Lock : 100,000,000+
AppLock : 100,000,000+
AVG AntiVirus 2018 for Android Security : 100,000,000+
Security Master - Antivirus, VPN, AppLock, Booster : 500,000,000+
Batt

In [37]:
#further investigate what is in productivity
for app in android_final:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '1,000,000,000+' or
                                     app[5] == '500,000,000+' or
                                     app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


From the analysis, it seems that there are dominating apps in some of these categories that drive up the number of installations. The rest of the categories that are popular among the users are:
- GAME
- PHOTOGRAPHY
- TRAVEL_AND_LOCAL
- VIDEO_PLAYERS
- NEWS_AND_MAGAZINES
- SHOPPING

## Conclusion

By comparing the results of both data sets, it seems that there are a few categories that are popular among users and have low level of competition in both App Store and Google Play, which can likely increase the opportunity of being discovered and downloaded by a user.

These  categories are:
1. Photo & video
2. Travel
3. Books / News and Magazines
4. Shopping

From the four categories above, Photo & Video category is recommended for the following reasons:
- **Capability of targeting both personal and professional market:**
With the evolution of social media for commercial purposes, there is a high demand for easily accessible tools for digital marketing materials. An intuitive and simple app is likely to be popular among small business owners or freelancers for simple editing tasks.

- **Need of developing in-app content:**
For the other categories, i.e., travel, books and magazines and shopping, the requirement of content development is high comparatively before the release of the app to the market. For photo & video apps, the content will likely be user-generated, which will only require the access to the photo albums of the app user.

- **Need of networking effect:**
Photo & video apps can be used individually where the usage of the app is likely related to user preference. For travel, books and magazines and shopping apps, networking effect could play a more important role by the number of users and number of partners available in the app.

In conclusion, Photo & Video category is recommended for development.